core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManager.scala

```diff
@@ -97,28 +106,37 @@ private[spark] class HadoopDelegationTokenManager(
       ThreadUtils.newDaemonSingleThreadScheduledExecutor("Credential Renewal Thread")
 
     val ugi = UserGroupInformation.getCurrentUser()
-    if (ugi.isFromKeytab()) {
+    val tgtRenewalTask = if (ugi.isFromKeytab()) {
       // In Hadoop 2.x, renewal of the keytab-based login seems to be automatic, but in Hadoop 3.x,
       // it is configurable (see hadoop.kerberos.keytab.login.autorenewal.enabled, added in
       // HADOOP-9567). This task will make sure that the user stays logged in regardless of that
       // configuration's value. Note that checkTGTAndReloginFromKeytab() is a no-op if the TGT does
       // not need to be renewed yet.
-      val tgtRenewalTask = new Runnable() {
+      new Runnable() {
         override def run(): Unit = {
           ugi.checkTGTAndReloginFromKeytab()
```

Comment:
When testing this I get a warning message "WARN UserGroupInformation: Not attempting to re-login since the last re-login was attempted less than 600 seconds before.." every minute (I am using the default value of spark.yarn.kerberos.relogin.period).

Reply:
From looking at the code it seems like a problem with your config. As far as I understand, you'd get this if your tgt was about to expire but `hadoop.kerberos.min.seconds.before.relogin` has not passed yet. That defaults to 60s (not 600), but it seems you've changed that value in your config.

Reply:
Thanks @vanzin for the detailed explanations.
After some additional investigation I found that if I compile Spark with Hadoop 3.1 the behavior is OK. I can still reproduce the issue I mentioned with the standard 2.7 version in my environment. It appears that  `hadoop.kerberos.min.seconds.before.relogin` is not available in Hadoop 2.7 and was introduced in 2.8?

Reply:
I was testing with Hadoop 2 and did not see the error you saw. But let me take another look at the code.

Reply:
I should clarify that the warning messages I reported are for the case where I use the TGT and --conf spark.kerberos.renewal.credentials=ccache rather than using keytab, apologies for the possible confusion this may have generated. 
Looking now at the code for UserGroupInformatio.reloginFromTicketCache I can see that it calls hasSufficientTimeElapsed which is responsible for generating the warning message in question, when users are trying to renew at a rate higher than a certain frequency. As you pointed out, with hadoop.kerberos.min.seconds.before.relogin set to default value of 60 we are OK as it matches the default for spark.kerberos.relogin.period, (but this requires HADOOP-7930, e.g. Hadoop version >= 2.8)).  
On a related topic, I can see that checkTGTAndReloginFromKeytab has a "silent" way of checking if the rate of request for renewal is higher than the threshold, so no warning are generated in this case.
Does this make sense and is it reproducible in you Hadoop 2.7 environment too?

Reply:
> for the case where I use the TGT and --conf spark.kerberos.renewal.credentials=ccache

Ah, that makes more sense. I'm testing on CDH, which is based on 2.6 but has a lot of patches including HADOOP-7930, so I don't see that message.

Anyway, since all Hadoop libs I looked at seem to take care of keeping the TGT up-to-date, I'll remove this code, and we can re-introduce it later if needed.